lxd/instance_post.go

```diff
@@ -206,6 +196,85 @@ func instancePost(d *Daemon, r *http.Request) response.Response {
 		}
 	}
 
+	// Run the cluster placement after potentially forwarding the request to another member.
```

Comment:
It looks like the changes to this function could probably be split into 2 logical changes:

1. Re-organisation to use the new helper functions and switching to using `targetMemberInfo` var.
2. Introducing of the cluster groups and placement script support.

It would be helpful if these could be split like that (or similar) as we need to move carefully here in order to keep things efficient and to ensure we are handling the different scenarios correctly.

When it lands as single big change like this it makes it more difficult to understand all the ramifications.

Reply:
I have split it it into two commits:

1. Restructuring (using central functions), simple scheduling and default behavior (least instances) including doc updates
2. Adding the scriptlet scheduler